src/integration-test/java/com/commercetools/sync/integration/services/impl/ShoppingListServiceImplIT.java

```diff
@@ -0,0 +1,265 @@
+        assertThat(cache).hasSize(0);
+
+        cache = shoppingListService.cacheKeysToIds(singleton(shoppingList.getKey())).toCompletableFuture().join();
+        assertThat(cache).hasSize(1);
+
+        cache = shoppingListService.cacheKeysToIds(emptySet()).toCompletableFuture().join();
+        assertThat(cache).hasSize(1);
+
+        assertThat(errorCallBackExceptions).isEmpty();
+        assertThat(errorCallBackMessages).isEmpty();
+    }
+
+    @Test
+    void cacheKeysToIds_WithCachedKeys_ShouldReturnCachedKeysWithoutRequest() {
+        final SphereClient spyClient = spy(CTP_TARGET_CLIENT);
+        final ShoppingListSyncOptions shoppingListSyncOptions =
+            ShoppingListSyncOptionsBuilder.of(spyClient)
+                                          .errorCallback((exception, oldResource, newResource, updateActions) -> {
+                                              errorCallBackMessages.add(exception.getMessage());
+                                              errorCallBackExceptions.add(exception.getCause());
+                                          })
+                                          .build();
+        final ShoppingListService shoppingListService = new ShoppingListServiceImpl(shoppingListSyncOptions);
+
+
+        Map<String, String> cache = shoppingListService.cacheKeysToIds(singleton(shoppingList.getKey()))
+                                                       .toCompletableFuture().join();
+        assertThat(cache).hasSize(1);
+
+        cache = shoppingListService.cacheKeysToIds(singleton(shoppingList.getKey()))
+                                   .toCompletableFuture().join();
+        assertThat(cache).hasSize(1);
+
+        verify(spyClient, times(1)).execute(any());
+        assertThat(errorCallBackExceptions).isEmpty();
+        assertThat(errorCallBackMessages).isEmpty();
+    }
+
+    @Test
+    void createShoppingList_WithValidShoppingList_ShouldCreateShoppingList() {
+        //preparation
+        ProductType productType = createProductType(PRODUCT_TYPE_RESOURCE_PATH, CTP_TARGET_CLIENT);
+        final ProductDraft productDraft = ProductDraftBuilder
+            .of(ResourceIdentifier.ofKey(productType.getKey()), LocalizedString.ofEnglish("newProduct"),
+                LocalizedString.ofEnglish("foo"),
+                ProductVariantDraftBuilder.of().key("foo-new").sku("sku-new").build())
+            .key("newProduct")
+            .build();
+        executeBlocking(CTP_TARGET_CLIENT.execute(ProductCreateCommand.of(productDraft)));
+        LineItemDraft lineItemDraft = LineItemDraftBuilder.ofSku("sku-new", Long.valueOf(1)).build();
+        final ShoppingListDraftDsl newShoppingListDraft =
+            ShoppingListDraftBuilder.of(LocalizedString.ofEnglish("new_name"))
+                                    .key("new_key")
+                                    .plusLineItems(lineItemDraft)
```

Comment:
Can you add text line item in your test scenario ?

Reply:
2365da1